content/docs/source-git/work-with-source-git/build-locally.md

```diff
@@ -0,0 +1,52 @@
+---
+title: "Build current checkout locally"
+date: 2021-01-27T23:59:59+01:00
+draft: false
+weight: 3
+---
+
+# Build your current checkout locally
+
+Once you are comfortable with your local changes, you can build them before
+pushing them out. Just make sure that all code changes are committed, otherwise
+packit won't be able to create patch files out of them.
+
+We are going to use packit to achieve such a thing, so please make sure you have it
+[installed locally](/docs/guide/#have-packit-tooling-installed-locally).
+
+The command to create source RPMs is called `srpm` and that's how you can create one:
+```
+$ packit srpm
+SRPM: /home/tt/g/systemd/systemd-stable/systemd-247.1-2.g68d22b32.fc33.src.rpm
+```
+
+Our changes worked! We can try building them, either in our local environment
+using packit - this implies that all build dependencies are installed:
+```
+$ packit local-build
+```
+
+...or in [mock](https://github.com/rpm-software-management/mock):
+```
+mock --rebuild -r fedora-rawhide-x86_64 ./systemd-247.1-2.g68d22b32.fc33.src.rpm
+```
+
+`-r` allows you [to pick a chroot]({{< ref "configuration.md#available-copr-build-targets" >}})
+of your choice and these are exactly the same which you can specify for builds
+in your `packit.yaml`.
+
+If the build is passing locally, it may be time to [propose your changes to
+dist-git]({{< ref "propose-to-dist-git.md" >}}).
+
+
+## How packit generates an SRPM from a source-git repo?
+
+These are the steps:
+
+1. Download archive specified in specfile's Source directive.
```

Comment:
This always confuses me... why isn't the archive created from the Git repo the packit command is called in?

Reply:
several projects have hidden scripts which generate release archives, other projects utilize complex machinery (autotools) to create archives

so, for simple projects we could generate archives from git, for all gcc's, kernels, mysql's, cockpits it's much easier to download the existing archive

the other problem is that it would a major difference between production dist-git builds which use actual release archives

Reply:
Could we just use the `create-archive` action for the cases when the archive creation is not a trivial `git archive`?

Reply:
yes, we could: I was actually thinking about something similar when I played with ruby: packit would download the archive by default and when create-archive is set, it would instead use those commands